### PR TITLE
exit with non-zero exit code on failures

### PR DESCRIPTION
**Pull request by escattone**

### CODE REVIEW

File: deployer/src/deployer/upload.py
Comment (peterbe) [inline deployer/src/deployer/upload.py:424]:
```suggestion
            f"There were {totals.failed} failures. Exiting with a non-zero exit code."
```

File: deployer/src/deployer/upload.py
Comment (peterbe) [inline deployer/src/deployer/upload.py:407]:
```suggestion
```
Not needed with the suggested `click.ClickException` suggestion below.

File: deployer/src/deployer/upload.py
Comment (peterbe) [inline deployer/src/deployer/upload.py:392]: What's done is done but I wonder if you ever want to or should use the progress bar. 
Even if was hacking on this on my laptop, I would probably prefer the no-progress where it just logs line after line. If I don't like I can pipe it to a file or something. 
In fact, for Yari, which also as a progress bar, I never use it because I've put `BUILD_NO_PROGRESSBAR=true` in my `.env`.

**Review by peterbe — APPROVED** [review]
I like my suggestion for including the count in the final exception error rather than having to explain it and relying on looking at the log line above. 
But I don't care strongly enough to not approve this :)

File: deployer/src/deployer/upload.py
Comment (escattone) [reply deployer/src/deployer/upload.py:392]: I do find it nice for local testing full-blown uploads of tens of thousands of files where I don't care about the details (other than errors).

**Comment by peterbe** [comment]
You already have my approval.

**Pull request merged**
